pycc/local.py

```diff
@@ -479,6 +479,94 @@ def MP2_LPNO(self):
             ediff = emp2 - elast
 
             print("MP2 Iter %3d: MP2 Ecorr = %.15f  dE = % .5E  rmsd = % .5E" % (niter, emp2, ediff, rmsd))
+    
+        def local_MP2_loop(self):
+        """
+        Performs the MP2 loop in the transformed local basis (PAO, PNO)
+
+        Attributes
+        ----------
+        Q: transform between canonical VMO and local spaces
+        L: transform between local and semicanonical local spaces
+        eps: semicananonical local energies
+
+        Notes
+        -----
+
+        """
+
+        o = slice(0,self.no)
+        v = slice(self.no, self, self.no + self.nv)
+
+        # initial guess amplitude in the approximated MO basis
+        eps_occ = np.diag(self.H.F)[o]
+        eps_vir = np.diag(self.H.F)[v]
+        Dijab = eps_occ.reshape(-1,1,1,1) + eps_occ.reshape(-1,1,1) - eps_vir.reshape(-1,1) - eps_vir
+        t2 = self.H.ERI[o,o,v,v]/Dijab
+
+        # Transformation basis for the virtual-virtual blocks of the Fock matrix, ERI tensor,
+        # T2 tensor, L(2g_ijab - gijba) tensor, and overlap matrix to PAO/PNO basis for each pair
+
+        # Transformed virtual-virtual block of the overlap matrix to the PAO/PNO
+        if self.local.upper() in ["LPAO"]:
+            S = self.L.T @ self.Q.T @ self.S[v,v] @ self.Q @ self.L
```

Comment:
I don't think `S` is stored on `self`; you will have to recreate it (see L91-L95 in `_build_PAO`)

Reply:
also, remember there is a `Q` and `L` for _every_ `ij` pair - the S matrix is NLO x NAO